SplunkRumWorkspace/SplunkRum/SplunkRum/Session.swift

```diff
@@ -52,16 +54,22 @@ func getRumSessionId() -> String {
             sessionIdLock.unlock()
         }
     }
-    if Date() > sessionIdExpiration {
+    if Date() > sessionIdExpiration { // expire in 4 hours
         sessionIdExpiration = Date().addingTimeInterval(TimeInterval(MAX_SESSION_AGE_SECONDS))
+        sessionIdInActivityExpiration = Date().addingTimeInterval(TimeInterval(MAXSESSIONINACTIVEAGESECONDS))
         rumSessionId = generateNewSessionId()
         callbacks = sessionIdCallbacks
+    } else if Date() > sessionIdInActivityExpiration { // expire 15 min
```

Comment:
There's nothing in here actually checking for inactivity.  This check should be done when the app is being brought back to the foreground, having been in the background for more than the `INACTIVE` timeout.  Such logic probably belongs in `AppLifecycle.swift` rather than here, with a method like `invalidateSession` placed here that the inactivity logic can call.